### PR TITLE
PHP 8.1 support in-initializers

### PR DESCRIPTION
**Pull request by kylekatarnls**
Type: feature
Issue: #556
Breaking change: no

### CODE REVIEW

**Comment by AJenbo** [comment]
Nice work, this is one of the best features of 8.1 :)

I'll try and review this either this weekend or the coming week.

If you need inspiration for what to work on next Enum support would be much appreciated :sweat_smile:

File: src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php [hunk 1]
```diff
@@ -7268,12 +7268,25 @@ protected function parseVariableDeclarator()
...
-            $declarator->setValue($this->parseStaticValueOrStaticArray());
+            $declarator->setValue($this->parseVariableDefaultValue());
         }
 
         return $this->setNodePositionsAndReturn($declarator);
     }
 
+    /**
+     * This method will parse a default value after a parameter/static variable/constant
+     * declaration.
```
Comment (ravage84) [inline src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:7279]:
```suggestion
     * Parse a default value after a parameter, static variable or constant declaration.
```

File: src/test/php/PDepend/Source/Language/PHP/Features/PHP81/InInitializersTest.php [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+use PDepend\AbstractTest;
+use PDepend\Source\AST\ASTAllocationExpression;
+use PDepend\Source\AST\ASTFormalParameter;
+use PDepend\Source\AST\ASTFormalParameters;
+use PDepend\Source\AST\ASTValue;
+use PDepend\Source\AST\ASTVariableDeclarator;
+
+/**
+ * @copyright 2008-2017 Manuel Pichler. All rights reserved.
```
Comment (ravage84) [inline src/test/php/PDepend/Source/Language/PHP/Features/PHP81/InInitializersTest.php:51]: Just a side note: We shouldn't attribute this to him anymore, as he actively left the project and he does not in any way do an want to contriibute to its new code.

File: src/test/resources/files/Source/Language/PHP/Features/PHP81/InInitializers/testInInitializers.php [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<?php
+class Foo
+{
+    public function __construct(
+        public Bar $bar = new Bar,
```
Comment (ravage84) [inline src/test/resources/files/Source/Language/PHP/Features/PHP81/InInitializers/testInInitializers.php:5]: I think it wouldn't hurt if we extended this example/test to two parameters, the second with a less open visiblity, may be.
May be even a third with no value.

**Review by ravage84 — CHANGES_REQUESTED** [review]

File: src/test/php/PDepend/Source/Language/PHP/Features/PHP81/InInitializersTest.php (hunk 2 above)
Comment (kylekatarnls) [inline src/test/php/PDepend/Source/Language/PHP/Features/PHP81/InInitializersTest.php:51]:
```suggestion
```

**Review by AJenbo — DISMISSED** [review]

File: src/test/resources/files/Source/Language/PHP/Features/PHP81/InInitializers/testInInitializers.php (hunk 3 above)
Comment (kylekatarnls) [reply src/test/resources/files/Source/Language/PHP/Features/PHP81/InInitializers/testInInitializers.php:5]: Done.

**Review by tvbeek — APPROVED** [review]

**Pull request merged**